connectors/services/consume.py

```diff
@@ -0,0 +1,152 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""
+Event loop
+
+- polls for work by calling Elasticsearch on a regular basis
+- instanciates connector plugins
+- mirrors an Elasticsearch index with a collection of documents
+"""
+import asyncio
+import os
+import time
+
+from connectors.byoc import BYOIndex, DataSourceError, PipelineSettings, SyncJobIndex
+from connectors.byoei import ElasticServer
+from connectors.es import DEFAULT_LANGUAGE, defaults_for
+from connectors.logger import logger
+from connectors.services.base import BaseService
+from connectors.source import DataSourceConfiguration, get_source_klass
+from connectors.utils import CancellableSleeps, trace_mem
+
+
+class JobService(BaseService):
+    def __init__(self, args):
+        super().__init__(args)
+        self.running = False
+
+        self._sleeps = CancellableSleeps()
+
+        elastic_config = self.config["elasticsearch"]
+
+        self.elastic_server = ElasticServer(elastic_config)
+        self.jobs = SyncJobIndex(elastic_config)
+        self.connectors = BYOIndex(elastic_config)
+
+        self.language_code = self.config.get(
+            "language_code", DEFAULT_LANGUAGE
+        )  # XXX: get language code from the connector instead
+        self.bulk_options = elastic_config.get("bulk", {})
+
+    async def run(self):
+        one_sync = self.args.one_sync
```

Comment:
I wonder if it is still useful to maintain the `one_sync` as of the args, with the decoupling of producer and consumer.

We want to have feature parity between ruby and python but we don't have this functionality in ruby. 

Another removable candidate is the `sync_now` flag. I do think it will be much simple and more readable if we make the app just do one thing

Reply:
For now I had to keep it the way it is - it's used within our CI to run functional tests.

You can for example run `NAME=mysql make ftest` and this command will attempt to run a one-shot sync. I don't like how it's done now - ideally I'd expose options to schedule a sync and to run a one-shot connector job against a target job as a separate CLI and then use this in our CI.

OR another option is in CI to start a server and push data into Kibana that will make the service run the sync, but it'll require a large rewrite of this as well.